crengine/src/private/lvfreetypeface.cpp

```diff
@@ -489,52 +494,54 @@ void LVFreeTypeFace::setFeatures(int features)
     _hash = 0; // Force lvstyles.cpp calcHash(font_ref_t) to recompute the hash
 }
 
-// Synthetized bold on a font that does not come with a bold variant.
-void LVFreeTypeFace::setEmbolden()
+// Synthetic thin/bold on a font that does not come with a corresponding variant.
+void LVFreeTypeFace::setSynthWeight(int synth_weight)
 {
-    _embolden = true;
-    // A real bold font has weight 700, vs 400 for the regular.
-    // LVFontBoldTransform did +200, so we get 600 (demibold).
-    // Let's do the same (even if I don't see why not +300).
-    _weight = (_weight + 200 > 900) ? 900 : _weight + 200;
+    if (_weight == synth_weight) {
+        _synth_weight = 0;
+        _synth_weight_half_strength = 0;
+        _synth_weight_advance_add = 0;
+        clearCache();
+        return;
+    }
+    _synth_weight = synth_weight;
     // And add +1 so we can know it's a fake/synthetized font, so we
     // can avoid getting it (and get the original regular font instead)
     // when synthetizing an other variant of that font.
-    _weight += 1;
-    // When not using Harfbuzz, we will simply call FT_GlyphSlot_Embolden()
-    // to get the glyphinfo and glyph with synthetized bold and increased
-    // metrics, and everything should work naturally:
+    _synth_weight += 1;
+    // We will simply call FT_Outline_Embolden()
+    // to get the glyphinfo and glyph with synthetized bold.
+    // To increase metrics, we add some embolding strength to glyph advance.
     //   "Embolden a glyph by a 'reasonable' value (which is highly a matter
     //   of taste) [...] For emboldened outlines the height, width, and
     //   advance metrics are increased by the strength of the emboldening".
-    //
-    // When using Harfbuzz, which uses itself the font metrics, that we
-    // can't tweak at all from outside, we'll get positioning based on
-    // the not-bolded font. We can't increase them as that would totally
-    // mess HB work.
```

Comment:
(Still in the process of updating all this for KOReader - posting some comments as a I read your stuff, before I can actually see how this does).
Are you sure this is all fine with Harfbuzz ? adding some advance just like that? How does it do with arabic cursive, diacritics, etc... ?

Reply:
> Are you sure this is all fine with Harfbuzz ? adding some advance just like that?

No, I unsure.

> How does it do with arabic cursive, diacritics, etc... ?

I haven't tested this at all. Somehow I didn't think about it.